CMakeLists.txt

```diff
@@ -16,7 +16,10 @@ set(CMAKE_CXX_STANDARD ${IRODS_CXX_STANDARD})
 set(CMAKE_CXX_EXTENSIONS OFF)
 # export-dynamic so stacktrace entries from executables have function names.
 set(CMAKE_EXE_LINKER_FLAGS_INIT "-rdynamic -Wl,--export-dynamic -Wl,--enable-new-dtags -Wl,--as-needed -Wl,-z,defs")
+# we have intentionally(?) undefined symbols in the plugins, so no -Wl,-z,defs
```

Comment:
Which symbols are undefined?

Reply:
`add_persistence_plugin`. It's defined in the executable, but left undefined in the plugins, so that the executable can provide the symbol for the plugins.